docs/source/routing.rst

```diff
@@ -0,0 +1,240 @@
+             task))
+
+So we create a utility function to find a task by its id. All this
+could just be an interface to a database.
+
+.. code-block:: common-lisp-repl
+
+   TODO> (defun get-task (id)
+           (gethash id *store*))
+
+When we render the tasks list, we add an href on the task, so we can go to ``/tasks/<id>``::
+
+.. code-block:: common-lisp-repl
+
+   TODO> (defmethod render ((task task))
+           (with-html
+             (:p (:input :type "checkbox"
+                         :checked (done task)
+                         :onclick (make-js-action
+                                   (lambda (&key &allow-other-keys)
+                                     (toggle task))))
+                 (:span (if (done task)
+                            (with-html
+                              (:s (title task)))
+                            (:a :href (format nil "/tasks/~a" (id task)) ;; <-- only addition.
+                                (title task)))))))
+
+
+The task-page widget
+====================
+
+In Weblocks, an HTML block that we want to display, and possibly update
+independently, is a widget. Here, we want to show a task's details on
+their own page, it is then a widget.
+
+.. code-block:: common-lisp-repl
+
+   TODO> (defwidget task-page ()
+           ((task
+             :initarg :task
+             :initform nil
+             :accessor task-page-task)))
+
+   TODO> (defmethod render ((task-page task-page))
+           (let ((task (task-page-task task-page)))
+             (with-html
+               (:div "Task " (id task))
+               (:h1 (title task))
+               (:div (if (done task) "Done!" "To Do."))
+               (:div "Lorem ipsum…"))))
+
+
+Defining routes
+===============
+
+At this point we can think of our routes like this::
+
+.. code-block:: common-lisp
+
+    (defroutes tasks-routes
+      ("/tasks/\\d+" <create the task-page widget>)
+      ("/tasks/" (make-task-list)))
+
+The regexp ``\\d+`` will capture any url that is formed of digits and
+contains at least one.
+
+As we see, the ``task-page`` constructor will need to get the id
+matched by the route.
+
+
+Path and url parameters
+=======================
+
+To get the current path, use ``(weblocks/request:get-path)``. Then,
+you can find the matching parameters with ``cl-ppcre``.
+
+Our ``task-page`` constructor becomes:
+
+.. code-block:: common-lisp-repl
+
+   TODO> (defun make-task-page ()
+           (let* ((path (weblocks/request:get-path))
+                  (id (first (ppcre:all-matches-as-strings "\\d+" path)))
+                  (task (get-task (parse-integer id)))
+             (if task
+                 (make-instance 'task-page :task task)
+                 (not-found))))
+   TODO> (defun not-found ()
+           "Show a 404 not found page."
+           (with-html
+             (:div "Task not found.")))
+
+And our router is simply::
+
+   TODO> (defroutes tasks-routes
+           ("/tasks/task/\\d+" (make-task-page))
```

Comment:
@vindarel In line 172 you defined the route as `"/tasks/\\d+"`. Coul be (just a little bit) confusing to finally define the route with a different path (`"/tasks/task/\\d+"`).

Reply:
damn, good catch thanks